### DISCUSSION TITLE
Tools to automate task through Shellhub

### DISCUSSION THREAD

**Discussion by benasse** [shellhub-io/shellhub#2853] [General]
Hi everyone,

I was wondering if anyone had tried using a automation tool through shellhub.

I think ansible is the most present on the market, but I couldn't get it to work with shellhub. 
I think this is because it uses unix socket forwarding and it's not supported at the moment.

To be able to automate tasks I tried another software called [SPOT](https://github.com/umputun/spot) which is nice.
This one is less complex than ansible and offers the same kind of automation functionality.
It uses a simple exec to run a command on the hosts, and sftp upload the file to the hosts.

For the moment I have succeeded in getting the executions to work with a simple command, however there seems to be a bug when transfers via sftp are used.

In your projects do you use tools to automate tasks through shellhub?

**Comment by gustavosbarreto** [comment]
> For the moment I have succeeded in getting the executions to work with a simple command, however there seems to be a bug when transfers via sftp are used.

In the next stable version, there will be improvements in file transfers via SFTP.

> In your projects do you use tools to automate tasks through shellhub?

I have been using n8n alongside ShellHub for personal projects. Moreover, I have been testing n8n on ShellHub's internal services to measure server latency and health.

**Reply by benasse** [reply]
> In the next stable version, there will be improvements in file transfers via SFTP.

Ok thank you for the information, I will test and give you feedback when this version is released or under development.

> I have been using n8n alongside ShellHub for personal projects. Moreover, I have been testing n8n on ShellHub's internal services to measure server latency and health.

Interesting, for my home automation I use node red, I'll see what can be done!
